test/rev.js

```diff
@@ -44,72 +44,78 @@ test('stores the hashes for later', async t => {
 	t.is(file.revHash, 'd41d8cd98f');
 });
 
-test.cb('handles sourcemaps transparently', t => {
+test('handles sourcemaps transparently', async t => {
 	const stream = rev();
-
-	stream.on('data', file => {
-		if (path.extname(file.path) === '.map') {
-			t.is(file.path, path.normalize('maps/pastissada-d41d8cd98f.css.map'));
-			t.end();
-		}
+	const data = pEventIterator(stream, 'data', {
+		resolutionEvents: ['finish'],
 	});
 
 	stream.write(createFile({
-		path: 'pastissada.css'
+		path: 'pastissada.css',
 	}));
 
 	stream.end(createFile({
 		path: 'maps/pastissada.css.map',
-		contents: JSON.stringify({file: 'pastissada.css'})
+		contents: JSON.stringify({file: 'pastissada.css'}),
 	}));
-});
 
-test.cb('handles unparseable sourcemaps correctly', t => {
-	const stream = rev();
-
-	stream.on('data', file => {
+	for await (const file of data) {
 		if (path.extname(file.path) === '.map') {
-			t.is(file.path, 'pastissada-d41d8cd98f.css.map');
-			t.end();
+			t.is(file.path, path.normalize('maps/pastissada-d41d8cd98f.css.map'));
 		}
+	}
+});
+
+test('handles unparseable sourcemaps correctly', async t => {
+	const stream = rev();
+	const data = pEventIterator(stream, 'data', {
+		resolutionEvents: ['finish'],
 	});
 
 	stream.write(createFile({
-		path: 'pastissada.css'
+		path: 'pastissada.css',
 	}));
 
 	stream.end(createFile({
 		path: 'pastissada.css.map',
-		contents: 'Wait a minute, this is invalid JSON!'
+		contents: 'Wait a minute, this is invalid JSON!',
 	}));
-});
-
-test.cb('okay when the optional sourcemap.file is not defined', t => {
-	const stream = rev();
 
-	stream.on('data', file => {
+	for await (const file of data) {
 		if (path.extname(file.path) === '.map') {
 			t.is(file.path, 'pastissada-d41d8cd98f.css.map');
-			t.end();
 		}
+	}
+});
+
+test('okay when the optional sourcemap.file is not defined', async t => {
+	const stream = rev();
+	const data = pEventIterator(stream, 'data', {
+		resolutionEvents: ['finish'],
 	});
 
 	stream.write(createFile({
-		path: 'pastissada.css'
+		path: 'pastissada.css',
 	}));
 
 	stream.end(createFile({
 		path: 'pastissada.css.map',
-		contents: JSON.stringify({})
+		contents: JSON.stringify({}),
 	}));
+
+	for await (const file of data) {
+		if (path.extname(file.path) === '.map') {
+			t.is(file.path, 'pastissada-d41d8cd98f.css.map');
+		}
+	}
```

Comment:
Do an assert for the correct file count too. Currently, if the stream never emits, it would not be caught and test would succeed.

Reply:
➡️ https://github.com/sindresorhus/gulp-rev/pull/267/commits/553466b0f75371af33a199c1f835e4ec6122124c